app/webpacker/components/ContactsPage/SubForms/Competition/index.jsx

```diff
@@ -3,25 +3,28 @@ import { FormTextArea } from 'semantic-ui-react';
 import WcaSearch from '../../../SearchWidget/WcaSearch';
 import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
 import I18n from '../../../../lib/i18n';
+import { useDispatch, useStore } from '../../../../lib/providers/StoreProvider';
+import { updateSubFormCompetition } from '../../store/actions';
+
+export default function Competition() {
+  const { competition } = useStore();
+  const dispatch = useDispatch();
+  const handleFormChange = (_, { name, value }) => dispatch(updateSubFormCompetition(name, value));
 
-export default function Competition({ formValues, setFormValues }) {
-  const handleFormChange = (_, { name, value }) => setFormValues(
-    { ...formValues, [name]: value },
-  );
   return (
     <>
       <WcaSearch
         label={I18n.t('page.contacts.form.competition.competition.label')}
         name="competition"
-        value={formValues.competition}
+        value={competition.competition}
```

Comment:
If we have `x.x` then I feel like one of the `x`s is probably not named well.

Reply:
You are right, once https://github.com/thewca/worldcubeassociation.org/pull/9280 is merged, I can rebase and change it to `competition.competitionId`.